pkg/adapter/store.go

```diff
@@ -7,15 +7,14 @@ import (
 	cmtstateproto "github.com/cometbft/cometbft/proto/tendermint/state"
 	cmtstate "github.com/cometbft/cometbft/state"
 	proto "github.com/cosmos/gogoproto/proto"
-
-	"github.com/rollkit/rollkit/pkg/store"
+	ds "github.com/ipfs/go-datastore"
 )
 
 const stateKey = "abci-s"
 
 // loadState loads the state from disk
-func loadState(ctx context.Context, s store.Store) (*cmtstate.State, error) {
-	data, err := s.GetMetadata(ctx, stateKey)
+func loadState(ctx context.Context, s ds.Batching) (*cmtstate.State, error) {
+	data, err := s.Get(ctx, ds.NewKey(stateKey))
```

Comment:
Any chance we can make a helper for this and should we store it under a new prefix key. Maybe abci prefix?